packages/plugins/Traderxyz/src/SNSAdaptor/PostPreview.tsx

```diff
@@ -0,0 +1,304 @@
+        color: theme.palette.mode === 'dark' ? '#ffffff' : '#000000',
+    },
+}))
+
+/**
+ * This page we use to full fill order
+ */
+
+export function PostPreview({ info }: { info: TradeMetaData }) {
+    const selectedChainId = useChainId()
+    const nftSwapSdk = useTraderApi(selectedChainId)
+    const account = useAccount()
+    const { classes } = useStyles()
+    const t = useI18N()
+    const theme = useTheme()
+    const { showSnackbar } = useCustomSnackbar()
+
+    const signOrder = useCallback(async () => {
+        const normalizeSignedOrder = nftSwapSdk.normalizeSignedOrder(info.signedOrder)
+        const walletAddressUserA = account
+        const assetsToSwapUserA = [info.assetsInfo.receiving_token, ...info.assetsInfo.nfts]
+
+        /// Check order status
+
+        // Invalid = 0,
+        // InvalidMakerAssetAmount = 1,
+        // InvalidTakerAssetAmount = 2,
+        // Fillable = 3,
+        // Expired = 4,
+        // FullyFilled = 5,
+        // Cancelled = 6
+
+        const orderStatus = await nftSwapSdk.getOrderStatus(normalizeSignedOrder)
+
+        if (orderStatus === 3) {
+            // Fillable
+            // Check if we need to approve the NFT for swapping
+            const approvalStatusForUserA = await nftSwapSdk.loadApprovalStatus(
+                assetsToSwapUserA[0] as SwappableAsset,
+                walletAddressUserA,
+            )
+
+            // Check if buyer needs to approve the sell token contract or no
+            if (!approvalStatusForUserA.contractApproved) {
+                // If not approved
+                const approvalTx = await nftSwapSdk.approveTokenOrNftByAsset(
+                    assetsToSwapUserA[0] as SwappableAsset,
+                    walletAddressUserA,
+                )
+
+                await approvalTx.wait().then(
+                    (msg) =>
+                        showSnackbar(
+                            `Approved ${assetsToSwapUserA[0]?.tokenAddress} contract to swap with 0x (txHash: ${msg.transactionHash})`,
+                            { variant: 'success' },
+                        ),
+                    (error) => showSnackbar(t.submit_order_submit_error_message() + error, { variant: 'error' }),
+                )
+            }
+            // fill order
+            const fillTx = await nftSwapSdk.fillSignedOrder(normalizeSignedOrder).then(
+                async (fillTx) => {
+                    const fillTxReceipt = await nftSwapSdk.awaitTransactionHash(fillTx?.hash).then(
+                        (fillTxReceipt) => {
+                            if (fillTxReceipt.status !== 0) {
+                                showSnackbar(
+                                    `\u{1F389} \u{1F973} ${t.submit_order_filled_message()} ${
+                                        fillTxReceipt?.transactionHash
+                                    }`,
+                                    { variant: 'success' },
+                                )
```

Comment:
plz extract the template string as variable.

```suggestion
                                const message = `\u{1F389} \u{1F973} ${t.submit_order_filled_message()} ${fillTxReceipt?.transactionHash}`
                                showSnackbar(message, { variant: 'success' })
```

Reply:
Fixed.